frontend/src/App.js

```diff
@@ -28,21 +28,32 @@ import { useUser } from './user';
  * @prop themeToggler - Function to toggle light/dark mode. Passed on to navbar and darkmodebutton
  */
 function App({ themeToggler }) {
+  // First load state
+  const [firstLoad, setFirstLoad] = useState(true);
   // Page initialized as loading
   const [loading, setLoading] = useState(true);
   // User context data
   const { user, userRefresh, fbRefresh } = useUser();
 
   // Refresh user worksheet and FB data on page load
   useEffect(() => {
-    const a = userRefresh(true);
-    const b = fbRefresh(true);
-
-    Promise.allSettled([a, b]).finally(() => {
-      // Set loading to false after user info and fb info is fetched
-      setLoading(false);
-    });
-  }, [userRefresh, fbRefresh]);
+    if (firstLoad) {
+      setFirstLoad(false);
+
+      const a = userRefresh(true);
+      const b = fbRefresh(true);
+
+      Promise.allSettled([a, b]).finally(() => {
+        // Set loading to false after user info and fb info is fetched
+        setTimeout(setLoading, 3000, false);
+      });
+    }
+  }, [userRefresh, fbRefresh, firstLoad]);
+
+  // If user is a first-year, redirect to beta
+  if (user.year === 2025) {
+    window.location.href = `https://beta.coursetable.com${window.location.pathname}`;
```

Comment:
idk if this is the best way to redirect, @kevinhu @maxyuan6717 what do you think?

Reply:
Not sure about this, maybe give react-router a shot? https://reactrouter.com/web/api/Redirect

Also, have you confirmed that this works by setting your own year to 2025 and visiting?

Reply:
yup it works! it seems like react router's Redirect component only works with paths on the same hostname URL..

Reply:
switched it to `window.open()`

Reply:
Just realized that there's another caveat here – if we only go by user year, we also include a bunch of grad school students as well as YC class of 2024 students on leave. Can you add conditions that the user has to be a YC student (should be the `school_code` attribute) and class of 2025 not on leave?

Reply:
Yeah sure i can filter out the grad students, but should we filter out the class of 24/25 on leave? since they prob havent used the main coursetable much either. also don’t know how we would filter them out..